jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/sync/SyncApplicationRunProfile.kt

```diff
@@ -0,0 +1,134 @@
+
+    inner class SyncApplicationRunProfileState(environment: ExecutionEnvironment) : CommandLineState(environment) {
+        val stackName = settings.stackName
+        override fun startProcess(): ProcessHandler {
+            val processHandler = KillableProcessHandler(getSamSyncCommand())
+            ProcessTerminatedListener.attach(processHandler)
+            return processHandler
+        }
+
+        private fun getSamSyncCommand(): GeneralCommandLine = getSamCli().apply {
+            withEnvironment(connection.toEnvironmentVariables())
+            withWorkDirectory(templatePath.toAbsolutePath().parent.toString())
+            addParameter("sync")
+            addParameter("--stack-name")
+            addParameter(stackName)
+            addParameter("--template-file")
+            addParameter(templatePath.toString())
+            addParameter("--s3-bucket")
+            addParameter(settings.bucket)
+            settings.ecrRepo?.let {
+                addParameter("--image-repository")
+                addParameter(it)
+            }
+            if (settings.capabilities.isNotEmpty()) {
+                addParameter("--capabilities")
+                addParameters(settings.capabilities.map { it.capability })
+            }
+            if (settings.parameters.isNotEmpty()) {
+                addParameter("--parameter-overrides")
+                settings.parameters.forEach { (key, value) ->
+                    addParameter(
+                        "${escapeParameter(key)}=${escapeParameter(value)}"
+                    )
+                }
+            }
+
+            if (settings.tags.isNotEmpty()) {
+                addParameter("--tags")
+                settings.tags.forEach { (key, value) ->
+                    addParameter(
+                        "${escapeParameter(key)}=${escapeParameter(value)}"
+                    )
+                }
+            }
+            addParameter("--no-dependency-layer")
+            if (syncOnlyCode) {
+                addParameter("--code")
+            }
+        }
+
+        private fun escapeParameter(param: String): String {
+            // Invert the quote if the string is already quoted
+            val quote = if (param.startsWith("\"") || param.endsWith("\"")) {
+                "'"
+            } else {
+                "\""
+            }
+
+            return quote + param + quote
+        }
+
+        override fun execute(executor: Executor, runner: ProgramRunner<*>) = super.execute(executor, runner).apply {
+            processHandler?.addProcessListener(object : ProcessAdapter() {
+
+                override fun onTextAvailable(event: ProcessEvent, outputType: Key<*>) {
+                    if (outputType === ProcessOutputTypes.STDOUT ||
+                        outputType === ProcessOutputTypes.STDERR
+                    ) {
+                        if (event.text.contains("[Y/n]:")) {
+                            insertAssertionNow = true
```

Comment:
needs to be declared in this instance of the process adapter, not in the companion